src/transformers/trainer.py

```diff
@@ -1846,12 +1852,12 @@ def training_step(self, model: nn.Module, inputs: Dict[str, Union[torch.Tensor,
         inputs = self._prepare_inputs(inputs)
 
         if is_sagemaker_mp_enabled():
-            scaler = self.scaler if self.use_amp else None
+            scaler = self.scaler if self.do_grad_scaling else None
             loss_mb = smp_forward_backward(model, inputs, self.args.gradient_accumulation_steps, scaler=scaler)
             return loss_mb.reduce_mean().detach().to(self.args.device)
 
         if self.use_amp:
-            with autocast():
+            with autocast(dtype=self.amp_dtype):
```

Comment:
I think this broke Trainer. `autocast()` doesn't have `dtype=self.amp_dtype` for older version I think.

@sgugger @stas00 @LysandreJik

Reply:
```
  File "/home/patrick/python_bin/transformers/trainer.py", line 1860, in training_step
    with autocast(dtype=self.amp_dtype):
TypeError: __init__() got an unexpected keyword argument 'dtype'
```

Reply:
Thank you, @patrickvonplaten - pushed a workaround here: https://github.com/huggingface/transformers/commit/14cc50d081c320331d850a64a54f1d732fa557ea